packages/expo-cli/src/credentials/api/ios.ts

```diff
@@ -17,256 +19,184 @@ type CredentialFields = {
+    if (!this.isPrefetched[accountName]) {
+      const credentials = await this.getAllCredentialsApi(accountName);
+      this.credentials[accountName] = {
+        appCredentials: keyBy(
+          credentials.appCredentials,
+          cred => `${cred.experienceName} ${cred.bundleIdentifier}`
+        ),
+        userCredentials: keyBy(credentials.userCredentials, cred => String(cred.id)),
+      };
+      this.isPrefetched[accountName] = true;
     }
-    return this.credentials;
+    return {
+      appCredentials: Object.values(this.credentials[accountName]?.appCredentials ?? {}),
+      userCredentials: Object.values(this.credentials[accountName]?.userCredentials ?? {}),
+    };
   }
 
-  async _fetchAllCredentialsApi(credentialLookupParameters: CredentialLookupParameters) {
-    return await this.api.getAsync('credentials/ios', credentialLookupParameters);
-  }
-  async _fetchAllCredentials() {
-    log('Fetching available credentials');
-    this.credentials = await this._fetchAllCredentialsApi({ owner: this.username });
-    this.shouldRefetch = false;
-  }
+  public async getDistCert(appLookupParams: AppLookupParams): Promise<IosDistCredentials | null> {
+    await this.ensureAppCredentials(appLookupParams);
+    const appCredentialsIndex = this.getAppCredentialsCacheIndex(appLookupParams);
+    const { accountName } = appLookupParams;
 
-  async getDistCert(
-    experienceName: string,
-    bundleIdentifier: string
-  ): Promise<IosDistCredentials | null> {
-    if (this.shouldRefetch) {
-      await this._fetchAllCredentials();
-    }
-    this._ensureAppCredentials(experienceName, bundleIdentifier);
-    const credIndex = this.credentials.appCredentials.findIndex(
-      app => app.experienceName === experienceName && app.bundleIdentifier === bundleIdentifier
-    );
-    const distCertExpoId = this.credentials.appCredentials[credIndex].distCredentialsId;
-    if (!distCertExpoId) {
+    const appCredentials = this.credentials[accountName]?.appCredentials?.[appCredentialsIndex];
+    if (!appCredentials || !appCredentials.distCredentialsId) {
       return null;
     }
-    const distCert = this.credentials.userCredentials.find(cred => cred.id === distCertExpoId) as
-      | IosDistCredentials
-      | undefined;
-    return distCert || null;
+
+    const distCert = this.credentials[accountName]?.userCredentials?.[
+      appCredentials.distCredentialsId
+    ] as IosDistCredentials | null;
+    return distCert ?? null;
   }
 
-  async _createDistCertApi(uploadUserCredentialParameters: UploadUserCredentialParameters) {
-    return await this.api.postAsync('credentials/ios/dist', uploadUserCredentialParameters);
+  public async createDistCert(
+    accountName: string,
+    credentials: appleApi.DistCert
+  ): Promise<IosDistCredentials> {
+    const id = await this.createDistCertApi(accountName, credentials);
+
+    // refetching because www might add some fields (certPrivateSigningKey or certSerialNumber)
```

Comment:
to clarify - `www` will extract the `certSerialNumber` from the provided dist cert, but it's not adding a `certPrivateSigningKey`. `certPrivateSigningKey` is generated via fastlane and given to www by the cli.